airflow/jobs/backfill_job.py

```diff
@@ -829,3 +831,78 @@ def _execute(self, session=None):
             executor.end()
 
         self.log.info("Backfill done. Exiting.")
+
+    @provide_session
+    def reset_state_for_orphaned_tasks(self, filter_by_dag_run=None, session=None):
+        """
+        This function checks if there are any tasks in the dagrun (or all)
+        that have a scheduled state but are not known by the
+        executor. If it finds those it will reset the state to None
```

Comment:
```suggestion
        that have a SCHEDULED state but are not known by the
        executor. If it finds those it will reset the state to NONE
```
What do you think?

Reply:
Oh yes, this is correct for this PR (I extracted it out of larger work where this function changes to only look at scheduled, bit not queued in the future l

Reply:
Oh, this wasn't even my words -- it was previously incorrect in BaseJob and I just moved it.

I will correct it anyway